packages/cdktf-cli/lib/cdktf-project.ts

```diff
@@ -527,24 +534,49 @@ export class CdktfProject {
       if (runningStacks.length >= maxParallelRuns) {
         await Promise.race(runningStacks.map((s) => s.currentWorkPromise));
       } else {
-        const nextRunningExecutor = await next();
-        if (!nextRunningExecutor) {
-          // In this case we have no pending stacks, but we also can not find a new executor
-          break;
+        try {
+          const nextRunningExecutor = await next();
+          if (!nextRunningExecutor) {
+            // In this case we have no pending stacks, but we also can not find a new executor
+            break;
+          }
+          method === "deploy"
+            ? nextRunningExecutor.deploy(refreshOnly)
+            : nextRunningExecutor.destroy();
+        } catch (e) {
+          // await next() threw an error because a stack failed to apply/destroy
+          // wait for all other currently running stacks to complete before propagating that error
+          logger.debug(
+            "Encountered an error while awaiting stack to finish",
+            e
+          );
+          logger.debug("Waiting for still running stacks to finish");
+          await Promise.allSettled(
+            this.stacksToRun
+              .filter((ex) => ex.currentWorkPromise)
+              .map((ex) => ex.currentWorkPromise)
+          );
+          logger.debug(
+            "Done waiting for still running stacks. All pending work finished"
+          );
+          throw e;
         }
-
-        method === "deploy"
-          ? nextRunningExecutor.deploy(refreshOnly)
-          : nextRunningExecutor.destroy();
       }
     }
 
     // We end the loop when all stacks are started, now we need to wait for them to be done
-    await Promise.all(
-      this.stacksToRun
-        .filter((ex) => ex.currentWorkPromise)
-        .map((ex) => ex.currentWorkPromise)
-    );
+    // We wait for all work to finish even if one of the promises threw an error.
+    const currentWork = this.stacksToRun
```

Comment:
I wonder if we can make this part a method, seems like we have a core function to execute and wait for all promises to settle in the event of an error